frontend.conf

```diff
@@ -31,6 +31,20 @@ server {
         
         access_log /var/log/nginx/access.log main_jwt;
     }
+
+    location = /login {
```

Comment:
This endpoint requirement was originally requested by PMs of NGINX Management Suite - API Connectivity Manager (a.k.a. NMS-ACM) as follows. In the meantime, I have added another common purpose.

**Product Requirements of NMS-ACM:**
- **Problem**: Current NJS implementation doesn’t have login endpoints for client apps (SPA) to interact with. Client Apps require /login function as part of relying party when user clicks on login button. 
- **Change Required**: Expose the `/login` nginx location block here (`openid_connect.server_conf`) & proxy it to existing authz endpoint configured in the map variable(`openid_connect_configuration.conf`) of  `$oidc_authz_endpoint`. This would outsource the login function to IdP as its configured. 

**Reason why we need regardless for the general purpose of NMS-ACM requirement:**
- The current example only shows the scenario where `/` location always need to start with user authentication.
- There are many web sites (which are proxied from `/` location) that do not have to be started with user authentication by using IdP's login GUI.
- For those cases, this endpoint can be used by the SPA when users click on login button.

**To Sum Up:**
- To prevent confusion with an additional example, I have removed this in this `frontend.conf`, and the endpoint has been moved into `openid_connect_configuration.conf` that can be used by customers based on their requirements.
- Let me know if this reason is not enough for you.

Reply:
The list of reasons is not very convincing. 
1. JWT authentication in NGINX works in such a way that if you specify `/login` location, then only it will be protected. The principle of "explicit allow" is a security principle that states that only actions or activities that have been explicitly permitted should be allowed to take place. This means that anything that is not specifically allowed is prohibited.
2. The `openid_connect.server_conf` file contains OIDC functional locations, that is, by default, the user does not need to change anything in it.
3. We work as a proxy, that is, the more locations we have, the less predictable the end application can be, simply because locations can overlap.

Your `/login` example is app specific, that is, nothing prevents you from creating the configuration that NMS-ACM need, we, in turn, pursue the goal of being as predictable (simple) as possible.
We don't make a Swiss Army knife for all occasions, we provide a simple solution that should meet the standard OIDC requirements and not be overly complicated.

Reply:
- FYI. **This feature was originally requested by the PM** in the team who **managed this OIDC repo** before you started working on this repo. Hence, it was firstly implemented into the NGINX Management Suite - API Connectivity Manager, and the PM's plan was to donate it here to be used by more customers.
- However, I totally understand what you point out, and we may **need to find another repo** to add **more app specific solutions**, because the purpose of this repo is just to provide a simple reference implementation. So we can provide much more convenient and easy solutions to customers to influence this OIDC solution over the world.
- Let me think of how to provide the simpler solution of the feature here. I appreciate your point.

Reply:
@route443 :
* The `/login` endpoint has been removed as this repo is only to provide reference implementation for a simple proxy.
* The `$oidc_landing_page` is only kept as I like your suggestion of variable name.
* The application level reference implementation will be separately considered in other repo as this repo is only for simple implementation for a proxy after discussing with the PM of NGINX Management Suite.
* Feel free to let me know if you have other thoughts.